examples/azure/terraform-azure-monitor-example/main.tf

```diff
@@ -0,0 +1,162 @@
+# See test/azure/terraform_azure_monitor_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "azurerm" {
+  version = "~> 2.20"
+
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy = true
+    }
+  }
+}
+
+# Configure the Microsoft Azure Active Directory Provider
+provider "azuread" {
+  version = "=0.7.0"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12.26"
+}
+
+resource "random_string" "short" {
+  length  = 3
+  lower   = true
+  upper   = false
+  number  = false
+  special = false
+}
+
+resource "random_string" "long" {
+  length  = 6
+  lower   = true
+  upper   = false
+  number  = false
+  special = false
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "monitor" {
+  name     = "terratest-monitor-rg-${var.postfix}"
+  location = var.location
+}
+
+data "azurerm_client_config" "current" {}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A STORAGE ACCOUNT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_storage_account" "monitor" {
+  name                     = format("%s%s", "storage", random_string.long.result)
+  resource_group_name      = azurerm_resource_group.monitor.name
+  location                 = azurerm_resource_group.monitor.location
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+
+  tags = {
+    environment = "staging"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A KEY VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_key_vault" "monitor" {
+  name                        = "kv-${var.postfix}"
+  location                    = azurerm_resource_group.monitor.location
+  resource_group_name         = azurerm_resource_group.monitor.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_enabled         = true
+  # soft_delete_retention_days  = 7
+  purge_protection_enabled = false
+
+  sku_name = "standard"
```

Comment:
is this the cheapest sku?  if so no problem, if not please adjust to cheapest.

Reply:
It is, yes.